### ISSUE TITLE
ValueError returned in jplhorizon when G is set to n.a.

### ISSUE THREAD

**Issue by drtgrav** [astropy/astroquery#1332]
When doing 

    sun2object_horizons = Horizons(id=hname,location=None,epochs=cls.jd).vectors()

excepts with this final error.
File "/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/astroquery/jplhorizons/core.py", line 1061, in _parse_horizons
    H = float(HGline[1].rstrip('G'))
ValueError: could not convert string to float: n.a.                 

I would suggest adding code to return a numpy.nan in this case rather than raising the error.

**Comment by bsipocz** [comment]
cc @mommermi

**Comment by bsipocz** [comment]
@drtgrav - could you please post a full example? It's very difficult to do any debugging with a single line full of otherwise unknown local variables.

**Comment by drtgrav** [comment]
```
Python 2.7.15 (v2.7.15:ca079a3ea3, Apr 29 2018, 20:59:26) 
[GCC 4.2.1 Compatible Apple LLVM 6.0 (clang-600.0.57)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> from astroquery import jplhorizons
>>> hname = "2010 NY104"
>>> jd = 2458000.
>>> jplhorizons.Horizons(id=hname,location=None,epochs=jd).vectors()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/astroquery/utils/class_or_instance.py", line 25, in f
    return self.fn(obj, *args, **kwds)
  File "/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/astroquery/utils/process_asyncs.py", line 29, in newmethod
    result = self._parse_result(response, verbose=verbose)
  File "/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/astroquery/jplhorizons/core.py", line 1225, in _parse_result
    data = self._parse_horizons(response.text)
  File "/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/astroquery/jplhorizons/core.py", line 1061, in _parse_horizons
    H = float(HGline[1].rstrip('G'))
ValueError: could not convert string to float: n.a.                 
>>> 
```

**Comment by mommermi** [comment]
@drtgrav note the new behavior: if `H` is not defined, it will not be listed in the resulting table, so you will have to check `'H' in eph` or something similar

**Comment by bsipocz** [comment]
Has been fixed by #1333
